src/transformers/models/cohere/modeling_cohere.py

```diff
@@ -216,13 +214,21 @@ def __init__(self, config: CohereConfig, layer_idx: Optional[int] = None):
         self.max_position_embeddings = config.max_position_embeddings
         self.rope_theta = config.rope_theta
         self.is_causal = True
+        self.use_qk_norm = config.use_qk_norm
 
         if (self.head_dim * self.num_heads) != self.hidden_size:
             raise ValueError(
                 f"hidden_size must be divisible by num_heads (got `hidden_size`: {self.hidden_size}"
                 f" and `num_heads`: {self.num_heads})."
             )
 
+        if self.use_qk_norm:
+            # When sharding the model using Tensor Parallelism, need to be careful to use n_local_heads
```

Comment:
Not sure if this comment is relevant as the model is not sharded by default

Reply:
Oh this is to warn others who port this to other frameworks from HF.